### PR TITLE
Cleaned up frontend TODOs

### PR DESCRIPTION
**Pull request by Puepis**
## Notion ticket link

https://www.notion.so/uwblueprintexecs/Review-remove-frontend-TODOs-dbd468a4bcf345c0b9ecaad1fe30e92b

## Implementation description

- addressed all TODOs
- display welcome modal for terms and conditions
- display resume
toc

## Steps to test

1. Go to login page and click on terms and conditions, verify that modal pops up
2. Login as a user and start a story test and upload resume
3. Sanity check: verify that changing name and stuff still works
3. Login as admin and verify that "Manage Tests" still fetches tests properly
4. Go to user's profile and verify that resume name is correct
resumefile

## What should reviewers focus on?

- no more frontend TODOs!!

## Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters: `docker exec -it planet-read_py-backend_1 /bin/bash -c "black . && isort --profile black ."` and I have generated new migrations: `flask db migrate -m "<your message>"`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/UserProfilePage.tsx [hunk 1]
```diff
@@ -270,24 +262,18 @@ const UserProfilePage = () => {
...
-          <Text>TODO</Text>
           {isAdmin && resumeId && (
             <>
               <Heading size="sm" marginTop="36px">
                 User Files
               </Heading>
-              {/* TODO: replace file name */}
               <Button variant="link" onClick={() => downloadFile(resumeId)}>
-                resume.pdf
+                {`${resumeFileName}.pdf`}
```
Comment (Puepis) [inline frontend/src/components/pages/UserProfilePage.tsx:271]: something i'm not sure about: if the user uploads their resume in `.docx` or some other format, the displayed name will be wrong (the file will still be downloaded correctly tho)

it's not that straightforward to get the file extension from the user profile page, so just curious how we should deal with this

Comment (gaoxk) [reply frontend/src/components/pages/UserProfilePage.tsx:271]: down to ommit .pdf from the name 👍

**Review by gaoxk — APPROVED** [review]
nice!
only nit i found (not introduced in this ticket) clicking any part of the box will impact the checkbox (including the link to open the terms and conditions). only clicking the checkbox should impact the checkbox
image

**Comment by Puepis** [comment]
> nice! only nit i found (not introduced in this ticket) clicking any part of the box will impact the checkbox (including the link to open the terms and conditions). only clicking the checkbox should impact the checkbox image

fixed!

**Pull request merged**